pkg/istio-agent/sds-agent.go

```diff
@@ -207,11 +203,7 @@ func NewAgent(proxyConfig *mesh.ProxyConfig, cfg *AgentConfig, sopts *security.O
 		sa.secOpts.TLSEnabled = false
 	}
 	// If proxy is using file mounted certs, JWT token is not needed.
-	if sa.secOpts.FileMountedCerts {
-		sa.secOpts.UseLocalJWT = false
-	} else {
-		sa.secOpts.UseLocalJWT = sa.CertsPath == "" // true if we don't have a key.pem
-	}
+	sa.secOpts.UseLocalJWT = !sa.secOpts.FileMountedCerts
```

Comment:
What happens in the other case, where CertsPath is specified ( i.e. we have certificates already - because it's a VM that is restarting ) ? I don't think in that case we are supposed to use local JWT.

Reply:
VM restarting is exactly the case I tested. In this case, we want to set UseLocalJWT to true even if certPath is specified. Otherwise, cert rotation with token will never happen.

Reply:
Sure, but does the cert rotation with existing cert still works ? Did you test that - and waited enough time for the token to expire, to confirm that expired token is not sent ? 

That was the point of the test there - if CertsPath != "", because we found a key.pem (previously provisioned) - don't send the token, which may be expired.

Reply:
The case where we want to use only token - and never the saved cert - should be covered by the new env you introduced.

Reply:
localJWT does not really control whether to use Cert or token for rotation. Using token or cert is controlled by the new env I added in this PR. @williamaronli is going to add tests for cert rotation, which should cover expired k8s token case.